<?xml version="1.0"?>
<!--Sensor Fusion-->

<launch>
    <!--tf-->
    <node pkg="tf" type="static_transform_publisher" name="velodyne2realsense"
          args="0.111393 -0.0103084 -0.0137764 0.0 0.0 0.0 velodyne camera_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="camera_link2camera_color_frame"
          args="0.000 0.015 0.000 0.001 -0.001 -0.017 camera_link camera_color_frame 100"/>
    <node pkg="tf" type="static_transform_publisher" name="camera_color_frame2camera_color_optical_frame"
          args="0.000 0.000 0.000 -1.571 0.000 -1.571 camera_color_frame camera_color_optical_frame 100"/>

    <!--velodyne-->
    <include file="$(find velodyne_pointcloud)/launch/32e_points.launch" />

    <!--republish-->
    <!-- <node name="republish" type="republish" pkg="image_transport" output="screen" -->
    <!--       args="compressed in:=/camera/color/image_raw raw out:=/camera/color/image_raw" /> -->

    <!--sensor fusion-->
    <node pkg="velodyne_camera_calibration" type="sensor_fusion" name="sensor_fusion" output="screen">
        <remap from="/image" to="/segmentation" />
        <!-- <remap from="/image" to="/camera/color/image_raw" /> -->
        <remap from="/cinfo" to="/camera/color/camera_info" />
        <remap from="/lidar" to="/velodyne_points" />
        <remap from="/colored_cloud" to="/velodyne_points/colored" />
        <remap from="/projection" to="/camera/color/projection" />
    </node>

    <!--segmentation-->
    <node pkg="semantic_segmentation" type="semantic_segmentation.py" name="semantic_segmentation">
    <remap from="/CompressedImage" to="/equirectangular/image_raw/compressed"/>
    <!-- <remap from="/CompressedImage" to="/camera/color/image_raw/compressed"/> -->
    </node>


    <!--rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find velodyne_camera_calibration)/config/sensor_fusion.rviz"/>

    <!--rosbag play-->
    <node pkg="rosbag" type="play" name="play" args="/home/amsl/bag_files/kakunin_theta_2022-09-17-15-19-28.bag" />
    <!-- <node pkg="rosbag" type="play" name="play" args="/home/amsl/bag_files/data.bag" /> -->

</launch>
